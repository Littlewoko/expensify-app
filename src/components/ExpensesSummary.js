import React from 'react'
import { connect } from 'react-redux'
import selectedExpenses from '../selectors/expenses'
import getExpensesTotal from '../selectors/expenses-total'
import numeral from 'numeral'

export const ExpensesSummary = ({ expenseCount, expenseTotal }) => {
    const expenseWord = expenseCount === 1 ? 'expense' : 'expenses'
    const formatTotal = numeral(expenseTotal / 100).format('0,0.00')
    return (
        <div>
            <h2>
                Viewing {expenseCount} {expenseWord} totalling Â£{formatTotal}    
            </h2>
        </div>
    )
}
    

const mapStateToProps = (state) => {
    const visibleExpenses = selectedExpenses(state.expenses, state.filters)
    return {
        expenseTotal: getExpensesTotal(visibleExpenses),
        expenseCount: visibleExpenses.length
    }
}

export default connect(mapStateToProps)(ExpensesSummary)