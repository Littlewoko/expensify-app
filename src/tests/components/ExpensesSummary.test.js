import React from 'react'
import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'
import { ExpensesSummary } from '../../components/ExpensesSummary'
import expenses from '../fixtures/expenses'

test('Should not render if no expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} expenseTotal={234} />)
    expect(toJSON(wrapper)).toMatchSnapshot()
}) 

test('Should render number of expenses and total value', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={3} expenseTotal={1234566} />)
    expect(toJSON(wrapper)).toMatchSnapshot()
}) 