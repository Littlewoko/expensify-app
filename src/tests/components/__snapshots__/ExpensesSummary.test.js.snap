// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should not render if no expenses 1`] = `
<div>
  <h2>
    Viewing 
    1
     
    expense
     totalling £
    2.34
  </h2>
</div>
`;

exports[`Should render number of expenses and total value 1`] = `
<div>
  <h2>
    Viewing 
    3
     
    expenses
     totalling £
    12,345.66
  </h2>
</div>
`;
